prober/http_test.go

```diff
@@ -97,6 +101,153 @@ func TestValidHTTPVersion(t *testing.T) {
+	type testdata struct {
+		msg                    string
+		contentLenght          int
+		uncompressedBodyLength int
+		handler                http.HandlerFunc
+		expectError            bool
+	}
+
+	testcases := map[string]testdata{
+		"brotli": func() testdata {
+			msg := "hello world"
+			var buf bytes.Buffer
+			fw := brotli.NewWriter(&buf)
+			fw.Write([]byte(msg))
+			fw.Close()
+			return testdata{
+				msg:                    msg,
+				contentLenght:          len(buf.Bytes()), // content lenght is the length of the compressed buffer
+				uncompressedBodyLength: len(msg),
+				handler: func(w http.ResponseWriter, r *http.Request) {
+					w.Header().Add("Content-Encoding", "br")
+					w.WriteHeader(http.StatusOK)
+					w.Write(buf.Bytes())
+				},
+			}
+		}(),
+
+		"deflate": func() testdata {
+			msg := "hello world"
+			var buf bytes.Buffer
+			// the only error path is an invalid compression level
+			fw, _ := flate.NewWriter(&buf, flate.DefaultCompression)
+			fw.Write([]byte(msg))
+			fw.Close()
+			return testdata{
+				msg:                    msg,
+				contentLenght:          len(buf.Bytes()), // content lenght is the length of the compressed buffer
+				uncompressedBodyLength: len(msg),
+				handler: func(w http.ResponseWriter, r *http.Request) {
+					w.Header().Add("Content-Encoding", "deflate")
+					w.WriteHeader(http.StatusOK)
+					w.Write(buf.Bytes())
+				},
+			}
+		}(),
+
+		"gzipped": func() testdata {
+			msg := "hello world"
+			var buf bytes.Buffer
+			gw := gzip.NewWriter(&buf)
+			gw.Write([]byte(msg))
+			gw.Close()
+			return testdata{
+				msg:                    msg,
+				contentLenght:          len(buf.Bytes()), // content lenght is the length of the compressed buffer
+				uncompressedBodyLength: len(msg),
+				handler: func(w http.ResponseWriter, r *http.Request) {
+					w.Header().Add("Content-Encoding", "gzip")
+					w.WriteHeader(http.StatusOK)
+					w.Write(buf.Bytes())
+				},
+			}
+		}(),
+
+		"identity": func() testdata {
+			msg := "hello world"
+			return testdata{
+				msg:                    msg,
+				contentLenght:          len([]byte(msg)),
+				uncompressedBodyLength: len(msg),
+				handler: func(w http.ResponseWriter, r *http.Request) {
+					w.Header().Add("Content-Encoding", "identity")
+					w.WriteHeader(http.StatusOK)
+					w.Write([]byte(msg))
+				},
+			}
+		}(),
+
+		"uncompressed": func() testdata {
+			msg := "hello world"
+			return testdata{
+				msg:                    msg,
+				contentLenght:          len(msg),
+				uncompressedBodyLength: len(msg),
+				handler: func(w http.ResponseWriter, r *http.Request) {
+					w.WriteHeader(http.StatusOK)
+					w.Write([]byte(msg))
+				},
+			}
+		}(),
+
+		"xxx": func() testdata {
+			msg := "hello world"
+			buf := []byte("xxx")
+			return testdata{
+				expectError:            true,
+				msg:                    msg,
+				contentLenght:          len(buf),
+				uncompressedBodyLength: len(buf), // compressed lenght is reported
```

Comment:
is this ok, or is it better to report -1?

Reply:
Hmm, hard to tell. We can't decode it so we can't know.